src/test/util/random.h

```diff
@@ -27,34 +27,10 @@ enum class SeedRand {
 /** Seed the RNG for testing. This affects all randomness, except GetStrongRandBytes(). */
 void SeedRandomForTest(SeedRand seed = SeedRand::SEED);
 
-static inline uint32_t InsecureRand32()
+template <RandomNumberGenerator Rng>
+inline CAmount RandMoney(Rng&& rng)
 {
-    return g_insecure_rand_ctx.rand32();
-}
-
-static inline uint256 InsecureRand256()
-{
-    return g_insecure_rand_ctx.rand256();
-}
-
-static inline uint64_t InsecureRandBits(int bits)
-{
-    return g_insecure_rand_ctx.randbits(bits);
-}
-
-static inline uint64_t InsecureRandRange(uint64_t range)
-{
-    return g_insecure_rand_ctx.randrange(range);
-}
-
-static inline bool InsecureRandBool()
-{
-    return g_insecure_rand_ctx.randbool();
-}
-
-static inline CAmount InsecureRandMoneyAmount()
-{
-    return static_cast<CAmount>(InsecureRandRange(MAX_MONEY + 1));
+    return CAmount{rng.randrange(MAX_MONEY + 1)};
```

Comment:
Just for my own understanding, the advantage here is now that any rng in a test can be used when calling RandMoney?

Reply:
Ah got it, makes sense. I need to check the individual commits. Thanks.